storybookjs/storybook
# UI: Remove `defaultProps` from `Link` component

Closes #24305

## What I did

In the already-closed issue referenced above, only a partial fix was originally applied.  This PR completes the entire fix by removing `defaultProps` from the `Link` typography component, as reinforced and confirmed by [this comment](https://github.com/storybookjs/storybook/issues/24305#issuecomment-1883319133) on the issue.  To summarize, this PR fixes the dev console warning `Warning: Link2: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.`

## Checklist for Contributors

### Testing

#### The changes in this PR are covered in the following automated tests:

This change should be functionally equivalent to its previous state (moving default prop values from the deprecated `defaultProps` object to the functional component's prop arguments).

#### Manual testing

Steps to reproduce the issue and verify the fix:
1. Pull down and switch to the `next` branch, which does not contain this fix.
2. Run `yarn task --task sandbox --start-from install --template nextjs/default-ts`. This will take a few minutes to run.  Once this completes, if the Storybook sandbox doesn't open in your browser, `cd` into `sandbox/nextjs-default-ts` and run `yarn storybook`.
3. In the Storybook sandbox that spun up, open your browser's dev console, and drill down to `Stories > renderers > react > decorators > Docs`.
4. Note the output in the dev console:  `Warning: Link2: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.`
5. Kill the server in the terminal and pull down and switch to the branch from this PR.
6. Repeat steps 2 and 3 above.
7. Note that this warning no longer appears in the dev console.

### Documentation

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
